komposable-architecture-test/src/main/java/com/toggl/komposable/test/TestStore.kt

```diff
@@ -0,0 +1,434 @@
+package com.toggl.komposable.test
+
+import com.toggl.komposable.architecture.Effect
+import com.toggl.komposable.architecture.ReduceResult
+import com.toggl.komposable.architecture.Reducer
+import com.toggl.komposable.architecture.Store
+import com.toggl.komposable.extensions.createStore
+import com.toggl.komposable.extensions.map
+import com.toggl.komposable.scope.DispatcherProvider
+import io.kotest.matchers.shouldBe
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.Channel.Factory.BUFFERED
+import kotlinx.coroutines.currentCoroutineContext
+import kotlinx.coroutines.flow.onCompletion
+import kotlinx.coroutines.flow.onStart
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.TestScope
+import kotlinx.coroutines.test.advanceTimeBy
+import kotlinx.coroutines.test.runCurrent
+import kotlinx.coroutines.yield
+import java.util.UUID
+import java.util.logging.Logger
+import kotlin.reflect.KProperty
+import kotlin.reflect.jvm.isAccessible
```

Comment:
I have been converting the code and tests for the Kotlin Multiplatform conversion, and it would be helpful if new tests and code did not make use of jvm specific libraries.  Is there any chance these could be converted to be pure kotlin?

Reply:
Hey @sveltema, this would limit functionalities in this repo.

Do you mean kotlin.reflect.jvm? I'm using that to check difference between private fields as well.
Kotest as well? (they use similar approach, just for the jvm).

We could try to extract portions of code that use these jvm libraries, making it easier to replace when porting. Wdyt?

Reply:
Also, are you suggesting we don't use java.* anything as well?

Reply:
Yes.  Usage of JVM specific code (which otherwise is not present in the library other than tests which up to now are fairly easy to alter for KMP) would require a choice of:
- Giving up on the discussed KMP version.
- Making a KMP fork and just not worrying about it.
- Refactoring or otherwise removal of platform dependent code from the KMP version
- Making the JVM dependent code only available on JVM platforms and just leaving it out for other platforms or making other platform dependent versions.